compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUPromoteMatmulToFitMMA.cpp

```diff
@@ -106,8 +97,48 @@ class LLVMGPUPromoteMatmulToFitMMAPass final
     });
 
     IRRewriter rewriter(ctx);
-    for (auto op : candidates) {
-      padWithZeroValue(rewriter, op, targetIterType, nofold);
+    for (linalg::LinalgOp op : candidates) {
+      SmallVector<int64_t> padMultiples(op.getNumLoops(), 1);
+      auto config = dyn_cast_or_null<IREE::GPU::LoweringConfigAttr>(
+          getLoweringConfig(op));
+      if (config) {
+        switch (targetDimensions) {
+        case LLVMGPUMatmulPadOption::ParallelDims:
+          padMultiples = config.getStaticTilingLevelSizes(
+              (unsigned)IREE::GPU::TilingLevel::Workgroup, op);
+          break;
+        case LLVMGPUMatmulPadOption::ReductionDims:
+          padMultiples = config.getStaticTilingLevelSizes(
+              (unsigned)IREE::GPU::TilingLevel::Reduction, op);
+          break;
+        default:
+          llvm_unreachable("Unexpected target dimensions");
+          break;
+        }
+      }
+
+      // Populate padding dimensions.
+      SmallVector<int64_t> paddingDimensions;
+      for (auto [idx, iter] : llvm::enumerate(op.getIteratorTypesArray())) {
+        if (iter == targetIterType) {
+          paddingDimensions.push_back(idx);
+        }
+      }
+
+      // Populate tile sizes. We pad to multiples of workgroup/reduction
+      // tile sizes based on the selected target tiling dimensions.
+      // This pass is ran after the select target tiling is done to pad
+      // all dimensions to the select tile sizes.
+      SmallVector<int64_t> padToMultipleOf;
+      padToMultipleOf.reserve(paddingDimensions.size());
```

Comment:
I think we don't need the `reserve` method for most of `SmallVector` usages because it already allocate a ~64 bytes buffer. IMO, it's good for `std::vector`, but it is unnecessary for `SmallVector`.